elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchSchemaFactory.java

```diff
@@ -56,15 +58,30 @@ public ElasticsearchSchemaFactory() {
 
     try {
       final String coordinatesString = (String) map.get("coordinates");
-      Preconditions.checkState(coordinatesString != null,
-          "'coordinates' is missing in configuration");
 
-      final Map<String, Integer> coordinates = mapper.readValue(coordinatesString,
-          new TypeReference<Map<String, Integer>>() { });
+      final String hostsString = (String) map.get("hosts");
 
-      // create client
-      final RestClient client = connect(coordinates);
+      final Set<HttpHost> hostSet = new LinkedHashSet<>();
+
+      if (coordinatesString != null) {
+        final Map<String, Integer> coordinates = mapper.readValue(coordinatesString,
```

Comment:
For coordinates can you log a `WARNING` message  that `hosts` configuration attribute is preferred and `coordinates` is deprecated ?

Reply:
We also need to mention it in release notes for next release (probably 1.22)

Reply:
Added warning message.

Since this change is backward compatible it should be pushed in previous releases also till release-1.19 so it will support production environment where most of ES is on HTTPS.

Reply:
We don't usually do back-porting. Next calcite release 1.22 should happen by end of this year